packages/uploadthing/src/internal/parser.ts

```diff
@@ -1,25 +1,63 @@
-import type { Json, MaybePromise } from "@uploadthing/shared";
+import type { v1 } from "@standard-schema/spec";
+import * as Schema from "effect/Schema";
+
+import type { Json } from "@uploadthing/shared";
 
 /**
  * TODO: Do we wanna support effect/schema parsers now??
  */
 
-// Don't want to use Zod cause it's an optional dependency
-export type ParseFn<TType> = (input: unknown) => MaybePromise<TType>;
+export type ParseFn<TType> = (input: unknown) => Promise<TType>;
```

Comment:
we've only ever publically supported Zod before which has had both `parse` and `parseAsync` methods so this could only potentially be breaking for people relying on internals which is fine